### ISSUE TITLE
Issue combining certain operations

### ISSUE THREAD

**Issue by bohnstingl** [torch-spyre/torch-spyre#637]
### What happened?

When adding the result of .mean(dim=-1, keep_dim=True) operation with a tensor of the same shape, an error can be observed. For example, the snippet below works for `cpu`, but fails for `spyre`. In the snippet below, `res` and `x2` have the same shape `(64, 1)`, but cannot be added.

```
import torch

def test(x1, x2):
    res = x1.mean(dim=-1, keepdim=True)
    res = res + x2
    return res

x1 = torch.randn(64, 4096, dtype=torch.float16)
x1 = x1.to("spyre")
x2 = torch.randn(64, 1, dtype=torch.float16)
x2 = x2.to("spyre")

compiled = torch.compile(test)
output = compiled(x1, x2)
print(output)
```

### What did you expect to happen?

The program should just run as on `cpu` without issues.

### How can we reproduce it?

See my code snippet above

### Any environmental details we need to know?

I am on commit `6d0735dec6fb2802ff667afd9c3c6b4ed3a4f9a8`

### Relevant log output

```Shell

```

**Comment by raghukiran1224** [comment]
Re-triage: **cannot reproduce — resolved.** Verified on pod rganti-spyre-main at torch-spyre `9be4e3d` (current main).

```
mean(dim=-1, keepdim=True) + tensor   →  matches CPU reference
```
The original repro (`x1.mean(dim=-1,keepdim=True) + x2` on (64,4096)+(64,1) fp16) now compiles and produces correct output on Spyre. Closing as fixed.
